src/containers/App.js

```diff
@@ -0,0 +1,98 @@
+// src/containers/App.js
+
+import React, { Component } from 'react'
+
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS } from '../constants';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+import Todos from '../components/Todos';
+import Todo from '../components/Todo';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+
+		// JSX supports passing in element arrays
+		/*
+		const array = [
+			<Todo label="First todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Second todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Third todo..." deleteHandler={this.handleTodoDelete} />
+		];
+		*/
+		this.filter = this.filter.bind(this);
+		this.handleTodoAdd = this.handleTodoAdd.bind(this);
+		this.handleTodoClear = this.handleTodoClear.bind(this);
+		this.handleTodoDelete = this.handleTodoDelete.bind(this);
+		this.handleTodoComplete = this.handleTodoComplete.bind(this);
+
+		this.state = {
+			lastIndex: 1,
+			show: ALL_TODOS,
+			items: []
+		};		
+	}
+
+	handleTodoAdd(e) {
```

Comment:
I would prefer to have this handler in the Header component.  That way you can have input logic handled in the same file as the input element itself.  The only time when data would need to flow outside of the component will be when the user hits ENTER.